app/src/main/java/net/programmierecke/radiodroid2/station/ItemAdapterIconOnlyStation.java

```diff
@@ -1,27 +1,62 @@
 package net.programmierecke.radiodroid2.station;
 
 import android.content.SharedPreferences;
+import android.util.Log;
 import android.util.TypedValue;
+import android.view.ContextMenu;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
+import androidx.annotation.NonNull;
 import androidx.fragment.app.FragmentActivity;
 import androidx.preference.PreferenceManager;
 import androidx.recyclerview.widget.ItemTouchHelper;
 import androidx.recyclerview.widget.RecyclerView;
 
+import com.github.zawadz88.materialpopupmenu.MaterialPopupMenu;
+
 import net.programmierecke.radiodroid2.R;
 import net.programmierecke.radiodroid2.Utils;
 import net.programmierecke.radiodroid2.service.PlayerServiceUtil;
 import net.programmierecke.radiodroid2.utils.RecyclerItemMoveAndSwipeHelper;
 import net.programmierecke.radiodroid2.utils.SwipeableViewHolder;
 
-public class ItemAdapterIconOnlyStation extends ItemAdapterStation {
+public class ItemAdapterIconOnlyStation extends ItemAdapterStation implements RecyclerItemMoveAndSwipeHelper.MoveAndSwipeCallback<ItemAdapterStation.StationViewHolder> {
+    private static final String TAG = "IconOnlyStation";
+    private static final long MIN_INTERVAL_BETWEEN_DRAG_AND_MENU_OPEN = 200;
+    private final double DISMISS_MENU_DRAG_THRESHOLD = 0.15;
+    private RecyclerItemMoveAndSwipeHelper<ItemAdapterStation.StationViewHolder> swipeAndMoveHelper = null;
+    private final long NEVER_IN_THE_FUTURE = Long.MAX_VALUE / 2;
+    private long timeLastDragEnded = 0;
+
+    @Override
+    public void onDragged(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, double dX, double dY) {
```

Comment:
I think you should move this callback to a separate class, it would prettier and make possible to reuse for normal stations list.

Reply:
OK, I think I understand.

Reply:
I meant a separate class, as in a composition pattern, not adding more inheritance. Sorry, I just hate inheritance.

Reply:
Sorry, I missed that. I'd also have to think about it.